core/src/test/java/com/linecorp/armeria/server/cors/CorsServerErrorHanderTest.java

```diff
@@ -0,0 +1,97 @@
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.server.HttpResponseException;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.HttpStatusException;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+public class CorsServerErrorHanderTest {
+    private static final ClientFactory clientFactory = ClientFactory.ofDefault();
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            final HttpService myService = (ctx, req) -> HttpResponse.of(HttpStatus.OK);
+
+            addCorsServiceWithException(sb, myService, "/cors_status_exception",
+                                        HttpStatusException.of(HttpStatus.INTERNAL_SERVER_ERROR));
+            addCorsServiceWithException(sb, myService, "/cors_response_exception",
+                                        HttpResponseException.of(
+                                        HttpResponse.of(HttpStatus.INTERNAL_SERVER_ERROR)));
+        }
+    };
+
+    private static void addCorsServiceWithException(ServerBuilder sb, HttpService myService, String pathPattern,
+                                                    Exception exception) {
+        sb.service(pathPattern, myService.decorate(
+        CorsService.builder("http://example.com")
+        .allowRequestMethods(HttpMethod.POST, HttpMethod.GET)
+        .allowRequestHeaders("allow_request_header")
+        .exposeHeaders("expose_header_1", "expose_header_2")
+        .preflightResponseHeader("x-preflight-cors", "Hello CORS")
+        .newDecorator())
+        .decorate((delegate, ctx, req) -> {
+            throw exception; }));
+    }
+
+    private static WebClient client() {
+        return WebClient.builder(server.httpUri()).factory(clientFactory).build();
+    }
+
+    private static AggregatedHttpResponse request(WebClient client, HttpMethod method, String path,
+                                                  String origin, String requestMethod) {
+        return client.execute(RequestHeaders.of(
+                method, path, HttpHeaderNames.ACCEPT, "utf-8", HttpHeaderNames.ORIGIN, origin,
+                HttpHeaderNames.ACCESS_CONTROL_REQUEST_METHOD, requestMethod)).aggregate().join();
+    }
+
+    private static AggregatedHttpResponse preflightRequest(WebClient client, String path, String origin,
+                                                   String requestMethod) {
+        return request(client, HttpMethod.OPTIONS, path, origin, requestMethod);
+    }
+
+    @ParameterizedTest
+    @CsvSource({ "/cors_status_exception", "/cors_response_exception" })
+    void testCorsHeaderWithException(String path) {
+        final WebClient client = client();
```

Comment:
We can just use the webClient from the server and remove `client()` method.
```suggestion
        final WebClient client = server.webClient();
```
Could you also run the formatter in this class?